compositions/ManageShip.tsx

```diff
@@ -0,0 +1,48 @@
+import { useContractTransaction, WalletCallStatus } from "hooks/useContractTx";
+import React, { useCallback, useContext } from "react";
+import { ContractContext } from "providerts/ContractProvider";
+
+interface Props {
+  id: string;
+}
+
+export const ManageShip: React.FC<Props> = ({ id }) => {
+  const { Ships, isReadOnly } = useContext(ContractContext);
+  const {
+    handleTx: handleHarvest,
+    txStatus: harvestStatus,
+    txInProgress: harvIP,
+  } = useContractTransaction(1);
+
+  const harvest = useCallback(async () => {
+    try {
+      if (!Ships || isReadOnly) {
+        throw new Error("Contract is not authorised");
+      }
+
+      await handleHarvest(Ships.harvest(id));
```

Comment:
do we need an approval between V2 and ships, or can it use public functions

Reply:
It can use public functions

Reply:
We need to approve the ships contract and islands contract as minters on the wood, iron and silver token contracts though I think that is separate.